control/statesp.py

```diff
@@ -207,12 +209,33 @@ def __init__(self, *args, **kw):
         D = _ssmatrix(D)
 
         # TODO: use super here?
-        LTI.__init__(self, inputs=D.shape[1], outputs=D.shape[0], dt=dt)
+        LTI.__init__(self, inputs=D.shape[1], outputs=D.shape[0])
         self.A = A
         self.B = B
         self.C = C
         self.D = D
 
+        # now set dt
+        if len(args) == 4: 
+            if 'dt' in kwargs:
+                dt = kwargs['dt']
+            elif self.is_static_gain(): 
+                dt = None
+            else:
+                dt = config.defaults['control.default_dt']
+        elif len(args) == 5:
+            dt = args[4]
+            if 'dt' in kwargs:
+                warn('received multiple dt arguments, using positional arg dt=%s'%dt)
+        elif len(args) == 1:
+            try:
+                dt = args[0].dt
+            except NameError:
+                if self.is_static_gain(): 
+                    dt = None
+                else:
+                    dt = config.defaults['control.default_dt']
```

Comment:
Does it ever happen that the passed `StateSpace` object (no other type allowed) has no `dt` argument? If so, it will rise an `AttributeError` instead of `NameError`

Reply:
For a unit test covering this, see test_copy_constructor_nodt() in https://github.com/python-control/python-control/pull/438/commits/ee21d548d10f93de96210ef4d18adeadf03ca3c4 (#438)